### PR TITLE
add new keyboard dynamis v2

### PR DESCRIPTION
**Pull request by bbrfkr**
## Description

Add "dynamis_v2" keyboard that is upgradable 65% row staggered keyboard integrated a trackball.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by waffle87** [comment]
It would be preferable to add this to the existing `keyboards/bbrfkr/dynamis` direction.
eg.
```shell
bbrfkr
└── dynamis
    ├── info.json # info that applies to both revisions
    ├── keymaps
    │   └── default
    │       └── keymap.c
    ├── rev1
    │   ├── info.json # rev1 specific info
    │   └── rules.mk
    └── rev2
        ├── info.json # rev2 specific info
        └── rules.mk
```

**Comment by bbrfkr** [comment]
@waffle87
Thanks for your review.
I want to separate all files, because rev1 and rev2 are not compatible completely. In this case, how shoud I do?

**Comment by waffle87** [comment]
If you don't mind me asking, what are the major differences? The code doesn't really indicate any
```diff
diff dynamis/config.h dynamis_v2/config.h
1c1
< /* Copyright 2022 bbrfkr
---
> /* Copyright 2023 bbrfkr
18a19,32
> /* rgb num */
> #define RGBLED_NUM 5
> #define RGBLIGHT_EFFECT_BREATHING
> #define RGBLIGHT_EFFECT_RAINBOW_MOOD
> #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
> #define RGBLIGHT_EFFECT_SNAKE
> #define RGBLIGHT_EFFECT_KNIGHT
> #define RGBLIGHT_EFFECT_CHRISTMAS
> #define RGBLIGHT_EFFECT_STATIC_GRADIENT
> #define RGBLIGHT_EFFECT_RGB_TEST
> #define RGBLIGHT_EFFECT_ALTERNATING
> #define RGBLIGHT_EFFECT_TWINKLE
>
> /* PMW3360 */
20a35
> #define PMW3360_INITIAL_CPI 2400
Only in dynamis: dynamis.c
Only in dynamis_v2: dynamis_v2.c
diff dynamis/info.json dynamis_v2/info.json
2c2
<     "keyboard_name": "dynamis",
---
>     "keyboard_name": "dynamis_v2",
4c4
<     "url": "https://github.com/bbrfkr/keyboards",
---
>     "url": "https://github.com/bbrfkr/dynamis-keyboard",
8c8
<         "pid": "0x0001",
---
>         "pid": "0x0002",
12,13c12,13
<         "cols": ["F0", "F1", "F4", "F5", "F6", "F7", "C7"],
<         "rows": ["B6", "B4", "D6", "D5", "D1", "C6", "B5", "D7", "D4", "D0"]
---
>         "cols": ["D0", "D1", "F6", "F7", "D5", "F4", "F5"],
>         "rows": ["D4", "D6", "D7", "D3", "D2", "C6", "B6", "B5", "B4", "C7"]
18c18
<             {"pin_a": "B7", "pin_b": "E6"}
---
>             {"pin_a": "F0", "pin_b": "F1"}
21,35d20
<     "rgblight": {
<         "led_count": 10,
<         "animations": {
<             "breathing": true,
<             "rainbow_mood": true,
<             "rainbow_swirl": true,
<             "snake": true,
<             "knight": true,
<             "christmas": true,
<             "static_gradient": true,
<             "rgb_test": true,
<             "alternating": true,
<             "twinkle": true
<         }
<     },
37c22
<         "pin": "D3"
---
>         "pin": "E6"
Common subdirectories: dynamis/keymaps and dynamis_v2/keymaps
diff dynamis/readme.md dynamis_v2/readme.md
4c4
< ![dynamis](https://i.imgur.com/GRPE5hzh.jpg)
---
> ![dynamis](https://i.imgur.com/Zs6MWPe.jpeg)
6c6
< Gasket mounted 65% row staggered keyboard integrated a trackball
---
> Upgradable 65% row staggered keyboard integrated a trackball
23c23
<     make bbrfkr/dynamis:default
---
>     make bbrfkr/dynamis_v2:default
27c27
<     make bbrfkr/dynamis:default:flash
---
>     make bbrfkr/dynamis_v2:default:flash
```

**Comment by bbrfkr** [comment]
The pin assignment is different. Therefore, rev1 and rev2 pcb are not compatible.

In this case, it is not needed that all files are separeted?

**Comment by waffle87** [comment]
As per my original comment, that would be exactly what
```shell
    └── rev2
        ├── info.json # rev2 specific info
```
would mean.

File: keyboards/bbrfkr/dynamis_v2/config.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
+
+/* PMW3360 */
+#define PMW33XX_CS_PIN SPI_SS_PIN
+#define POINTING_DEVICE_INVERT_Y
+#define PMW3360_INITIAL_CPI 2400
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/config.h:35]: This handles this in the core code:
```suggestion
#define PMW33XX_CPI 2400U
```

File: keyboards/bbrfkr/dynamis_v2/dynamis_v2.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        tap_code(layer_state == 0 ? KC_WH_U : KC_VOLU);
+    }
+    return true;
+}
+#endif
+
+void pointing_device_init_kb(void) {
+    pointing_device_set_cpi(PMW3360_INITIAL_CPI);
+}
+
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/dynamis_v2.c:34]:
```suggestion
```

File: keyboards/bbrfkr/dynamis_v2/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# dynamis
+
+## general information
+![dynamis](https://i.imgur.com/Zs6MWPe.jpeg)
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/readme.md:4]:
```suggestion
![dynamis](https://i.imgur.com/Zs6MWPeh.jpeg)
```

File: keyboards/bbrfkr/dynamis_v2/rules.mk
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/rules.mk:1]: Most of this can, and should be migrated to info.json

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bbrfkr/dynamis_v2/config.h (hunk 1 above)
Comment (bbrfkr) [reply keyboards/bbrfkr/dynamis_v2/config.h]: Thanks. I fixed it.

File: keyboards/bbrfkr/dynamis_v2/dynamis_v2.c (hunk 2 above)
Comment (bbrfkr) [reply keyboards/bbrfkr/dynamis_v2/dynamis_v2.c]: OK. I removed it.

File: keyboards/bbrfkr/dynamis_v2/readme.md (hunk 3 above)
Comment (bbrfkr) [reply keyboards/bbrfkr/dynamis_v2/readme.md]: I fixed it.

File: keyboards/bbrfkr/dynamis_v2/rules.mk
Comment (bbrfkr) [reply keyboards/bbrfkr/dynamis_v2/rules.mk:1]: OK. I'll try it.

**Comment by bbrfkr** [comment]
I want to separate dynamis.c between v1 and v2. Could you tell me how it be able to be realized?

**Comment by drashna** [comment]
> I want to separate dynamis.c between v1 and v2. Could you tell me how it be able to be realized?

name the folders dynamis/rev1 and dynamis/rev2, rather than what you have here.

**Comment by bbrfkr** [comment]
I see. I try it! Thanks.

File: keyboards/bbrfkr/dynamis_v2/config.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#define RGBLIGHT_EFFECT_BREATHING
+#define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#define RGBLIGHT_EFFECT_SNAKE
+#define RGBLIGHT_EFFECT_KNIGHT
+#define RGBLIGHT_EFFECT_CHRISTMAS
+#define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#define RGBLIGHT_EFFECT_RGB_TEST
+#define RGBLIGHT_EFFECT_ALTERNATING
+#define RGBLIGHT_EFFECT_TWINKLE
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/config.h:30]: These should be moved to info.json. 

`qmk migrate` can help with that, as can `qmk info -f json`

File: keyboards/bbrfkr/dynamis_v2/dynamis_v2.c [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    if (clockwise) {
+        tap_code(layer_state == 0 ? KC_WH_D : KC_VOLD);
+    } else {
+        tap_code(layer_state == 0 ? KC_WH_U : KC_VOLU);
+    }
+    return true;
+}
+#endif
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/dynamis_v2.c:29]: This is very encoder map type behavior.  Keymaps should use it.

File: keyboards/bbrfkr/dynamis_v2/rules.mk [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = no
+POINTING_DEVICE_ENABLE = yes
+POINTING_DEVICE_DRIVER = pmw3360
+ENCODER_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/bbrfkr/dynamis_v2/rules.mk:17]: Most of these should be moved to info.json.

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**